test/sharness/t0082-repo-gc-auto.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/sh
+#
+# MIT Licensed; see the LICENSE file in this repository.
+#
+
+test_description="Test ipfs repo auto gc"
+
+. lib/test-lib.sh
+
+check_ipfs_storage() {
+    ipfs config Datastore.StorageMax
+}
+
+test_init_ipfs
+
+test_expect_success "generate 2 600 kB files and 2 MB file using go-random" '
+    random 600k 41 >600k1 &&
+    random 600k 42 >600k2 &&
+    random 2M 43 >2M
+'
+
+test_expect_success "set ipfs gc watermark, storage max, and gc timeout" '
+    test_config_set Datastore.StorageMax "2MB" &&
+    test_config_set --json Datastore.StorageGCWatermark 60 &&
+    test_config_set Datastore.GCPeriod "20ms"
+'
+
+test_launch_ipfs_daemon
+
+test_expect_success "adding data below watermark doesn't trigger auto gc" '
+    ipfs add 600k1 >/dev/null &&
+    disk_usage "$IPFS_PATH/blocks" >expected &&
+    go-sleep 40ms &&
+    disk_usage "$IPFS_PATH/blocks" >actual &&
+    test_cmp expected actual
+'
+
+#todo: `ipfs pin rm -r` is unclean so cant use du -sb "$IPFS_PATH/blocks" | awk "{print \$1}" >expected &&
+test_expect_success "adding data beyond watermark triggers auto gc" '
+    if test $(uname -s) = "Darwin"; then
+        expected=1000
+    else
+        expected=1000000
+    fi &&
+    HASH=`ipfs add -q 600k2` &&
+    ipfs pin rm -r $HASH &&
+    go-sleep 40ms &&
+    actual=$(disk_usage "$IPFS_PATH/blocks") &&
+    test "$actual" -lt "$expected"
+'
+
+test_expect_success "adding data beyond storageMax fails" '
+    test_must_fail ipfs add 2M 2>add_fail_out
+'
+
+test_expect_success "ipfs add not enough space message looks good" '
+    echo "Error: file size exceeds slack space allowed by storageMax. Maybe unpin some files?" >add_fail_exp &&
+    test_cmp add_fail_exp add_fail_out
+'
```

Comment:
i still think we need a test where we add (+unpin) files in a loop through a few (20?) GCs. just to make sure GC works over many times of use.